lib/board.rb

```diff
@@ -57,20 +57,35 @@ def check_validation(ship, coordinates)
   end
 
   def letters(coordinates)
-    (coordinates[0][0]..coordinates[-1][0]).to_a
+    [*(letters_sort(coordinates)[0]..letters_sort(coordinates)[-1])]
+  end
+
+  def letters_sort(coordinates)
+    coordinates.map do |coordinate|
+      coordinate[0]
+    end.uniq.sort
   end
 
   def numbers(coordinates)
-    num = coordinates.map do |coordinate|
-      coordinate[1]
-    end.sort
-    (num[0]..num[-1]).to_a
+    [*(numb_sort(coordinates)[0]..numb_sort(coordinates)[-1])]
+  end
+
+  def numb_sort(coordinates)
+    nums = []
+    coordinates.each do |coordinate|
```

Comment:
This looks great. Could you consolidate a litte code by doing 
`    nums = coordinates.map do |coordinate|
      if coordinate.chars.count == 2
        coordinate[1].to_i
      elsif coordinate.chars.count > 2
        coordinate[1..-1].to_i
      end
    end`
instead of 
`    nums = []
    coordinates.each do |coordinate|
      if coordinate.chars.count == 2
        nums << coordinate[1].to_i
      elsif coordinate.chars.count > 2
        nums << coordinate[1..-1].to_i
      end
    end`

Reply:
Done! Definitely cleaner with map.